src/data_provider/CMakeLists.txt

```diff
@@ -93,12 +93,23 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
       "${CMAKE_SOURCE_DIR}/src/packages/rpm*.cpp")
   add_definitions(-DLINUX_TYPE=LinuxType::STANDARD) # Standard compilation in compatible systems
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-  file(GLOB SYSINFO_SRC
-      "${CMAKE_SOURCE_DIR}/src/*Mac.cpp"
-      "${CMAKE_SOURCE_DIR}/src/*CommonBSD.cpp"
-      "${CMAKE_SOURCE_DIR}/src/packages/*Mac.cpp"
-      "${CMAKE_SOURCE_DIR}/src/network/*BSD.cpp"
-      "${CMAKE_SOURCE_DIR}/src/osinfo/sysOsParsers.cpp")
+  if(${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "arm64.*|ARM64.*")
+    file(GLOB SYSINFO_SRC
+        "${CMAKE_SOURCE_DIR}/src/*Mac.cpp"
+        "${CMAKE_SOURCE_DIR}/src/*CommonBSD.cpp"
+        "${CMAKE_SOURCE_DIR}/src/packages/*Mac.cpp"
+        "${CMAKE_SOURCE_DIR}/src/network/*BSD.cpp"
+        "${CMAKE_SOURCE_DIR}/src/osinfo/sysOsParsers.cpp"
+        "${CMAKE_SOURCE_DIR}/src/hardware/*ARMMac.cpp")
+  else()
+    file(GLOB SYSINFO_SRC
+        "${CMAKE_SOURCE_DIR}/src/*Mac.cpp"
+        "${CMAKE_SOURCE_DIR}/src/*CommonBSD.cpp"
+        "${CMAKE_SOURCE_DIR}/src/packages/*Mac.cpp"
+        "${CMAKE_SOURCE_DIR}/src/network/*BSD.cpp"
+        "${CMAKE_SOURCE_DIR}/src/osinfo/sysOsParsers.cpp"
+        "${CMAKE_SOURCE_DIR}/src/hardware/*X86_64Mac.cpp")
+  endif()
```

Comment:
CMake "list" command manages string values just like variables. But "file" command treats them as filepaths. For example, if a path with a wildcard (*) is used in the expression, then the output variable is expanded to all existing files that match the wildcard.
https://cmake.org/cmake/help/latest/command/file.html#filesystem
Because of this, I suppose that the command "file" is better.